### PR TITLE
Remove cross compiling for netcore50 

### PR DESCRIPTION
**Pull request by natemcmaster**
Add custom packing build step so tools can still be installed to UWP apps.

### CODE REVIEW

**Comment by natemcmaster** [comment]
cc @pranavkm for the makefile stuff
cc @divega @Eilon once this is in, we can remove netcore50 from EF's dependencies

https://github.com/aspnet/Common/compare/namc/no-netcore50
https://github.com/aspnet/DependencyInjection/compare/namc/no-netcore50
https://github.com/aspnet/Caching/compare/namc/no-netcore50
https://github.com/aspnet/PlatformAbstractions/compare/namc/no-netcore50
https://github.com/aspnet/Microsoft.Data.Sqlite/compare/namc/no-netcore50
https://github.com/aspnet/Logging/compare/namc/no-netcore50
https://github.com/aspnet/Options/compare/namc/no-netcore50

File: makefile.shade [hunk 1]
```diff
@@ -31,4 +34,50 @@ k-standard-goals
...
+      Log.Info("Repacking Nupkg: " + projectNupkg);
+
+       var extractToDirectory = projectNupkg + "-temp";
+      ZipFile.ExtractToDirectory(projectNupkg, extractToDirectory);
+
+      var nuspecFile = Files.Include(Path.Combine(extractToDirectory, "*.nuspec")).First();
+      var nuspec = new List<string>(File.ReadAllLines(nuspecFile));
+      var dependencies = Array.FindIndex<string>(nuspec.ToArray(), l => l.Contains("<dependencies>")) + 1;
+
+      var uap = nuspec.SkipWhile(l => !l.Contains("<group targetFramework=\".NETFramework4.5.1"))
```
Comment (pranavkm) [inline makefile.shade]: Do you want to make this a tool instead? Might be more reliable to use nuget APIs to do this. Here's what we do in Coherence-Signed: https://github.com/aspnet/Coherence-Signed/blob/dev/src/PushCoherence/Program.cs#L90-L122

**Comment by divega** [comment]
@natemcmaster can we chat a bit about the implications of this tomorrow?

File: makefile.shade (hunk 1 above)
Comment (natemcmaster) [inline makefile.shade]: Yes, this plus what we add in https://github.com/aspnet/EntityFramework/pull/5399 make me think we'd be better of making this a tool. Too bad we can't use nuspec directly or avoid compilation and yes still produce a nupkg that targets netcore50 (https://github.com/dotnet/cli/issues/897).

**Comment by natemcmaster** [comment]
@divega of course, i'll be here all day :)

**Comment by natemcmaster** [comment]
:watch: waiting to merge...found a potential show-stopper bug in .NET Native compiler + CoreFX. If this bug won't be fixed for RTM, then we'll need to continue cross compiling to old versions of CoreFX for UWP.

**Comment by natemcmaster** [comment]
cc @bricelam

**Comment by natemcmaster** [comment]
The corefx bug will be fixed for RTM. We can move ahead with this PR now.

**Comment by bricelam** [comment]
:shipit:

**Pull request merged**